File: blaze_flutter_sdk/ios/Sources/Ads/BlazeFlutterContentExtraInfo.swift
//
//  BlazeFlutterContentExtraInfo.swift
//  Pods
//
//  Created by Reuven Levitsky on 17/06/2025.
//

import BlazeSDK

public struct BlazeFlutterContentExtraInfo: Encodable {
    let previous: [String: String]?
    let current: [String: String]?
    let next: [String: String]?
}

public extension BlazeContentExtraInfo {
    
    var toFlutterModel: BlazeFlutterContentExtraInfo {
        return .init(
            previous: previous,
            current: current,
            next: next
        )
    }
    
}

File: blaze_flutter_sdk/android/src/main/kotlin/com/blaze/flutterblazesdk/ads/BlazeFlutterContentExtraInfo.kt
package com.blaze.flutterblazesdk.ads

import androidx.annotation.Keep
import com.blaze.blazesdk.ads.models.ui.BlazeContentExtraInfo

@Keep
data class BlazeFlutterContentExtraInfo(
        val previous: Map<String, String>?,
        val current: Map<String, String>?,
        val next: Map<String, String>?,
)

@Keep
fun BlazeContentExtraInfo.toFlutterModel(): BlazeFlutterContentExtraInfo {
        return BlazeFlutterContentExtraInfo(
                previous = this.previous,
                current = this.current,
                next = this.next
        )
}
